src/Runtime/ExecutionSession.hpp

```diff
@@ -24,15 +24,21 @@
 
 namespace onnx_mlir {
 
-typedef OMTensorList *(*entryPointFuncType)(OMTensorList *);
-typedef const char *(*signatureFuncType)();
-
+using entryPointFuncType = OMTensorList *(*)(OMTensorList *);
+using queryEntryPointsFuncType = const char **(*)();
+using signatureFuncType = const char *(*)(const char *);
 using OMTensorUniquePtr = std::unique_ptr<OMTensor, decltype(&omTensorDestroy)>;
 
 class ExecutionSession {
 public:
-  ExecutionSession(std::string sharedLibPath);
-  ExecutionSession(std::string sharedLibPath, std::string entryPointName);
+  ExecutionSession(std::string sharedLibPath, bool defaultEntryPoint = true);
+
+  // Get a NULL-terminated array of entry point names.
+  // For example {"run_addition, "run_substraction", NULL}
+  const std::string *queryEntryPoints() const;
+
+  // Set entry point for this session.
```

Comment:
Please just add in the comments that this call is necessary when the defaultEntryPoint is not used, or if it was used but the user wants now to use a different entry point.  

I have a question related to the string passed to the setEntryPoint. I know that for the "run" part, a dlsym is used, so the string chars are used to locate the symbol. But for the signatures, is it the address of the string that needs to be the same as recorded, or is it the content of the string?

In other words, are either of these calls working?

``` C++
setEntryPoint(queryEntryPoints()[0]); // use string directly
setEntryPoint(string(queryEntryPoints()[0])); // use a copy of the string
```

Reply:
> setEntryPoint(queryEntryPoints()[0]); // use string directly
> setEntryPoint(string(queryEntryPoints()[0])); // use a copy of the string
> is it the content of the string?

Yes, it is the string content. so both work.

Reply:
> Please just add in the comments that this call is necessary when the defaultEntryPoint is not used, or if it was used but the user wants now to use a different entry point.

Done this. Thanks!